packages/shorebird_cli/lib/src/shorebird_config_mixin.dart

```diff
@@ -105,4 +106,16 @@ app_id:
 
     pubspecFile.writeAsStringSync(editor.toString());
   }
+
+  Future<String> getShorebirdFlutterRevision() async {
+    final result = await process.run(
+      'git',
+      ['rev-parse', 'HEAD'],
+      workingDirectory: ShorebirdEnvironment.flutterDirectory.path,
+    );
+    if (result.exitCode != 0) {
+      throw Exception('Unable to determine flutter revision: ${result.stderr}');
+    }
+    return (result.stdout as String).trim();
+  }
```

Comment:
We have similar logic in shorebird/packages/shorebird_cli/lib/src/validators/shorebird_flutter_validator.dart. I think this implementation is a bit more robust, so we probably want to find a way to use this there.

Reply:
Are you referring to `_getFlutterVersion`? I saw it was similar but it seemed to be version-based instead of hash-based.